src/globals.c

```diff
@@ -1381,36 +1381,91 @@ void fatal(char *msg)
 /* Reports an error and specifying a position */
 void error(char *msg)
 {
-    /* Construct error source diagnostics, enabling precise identification of
-     * syntax and logic issues within the code.
-     */
-    int offset, start_idx, i = 0;
-    char diagnostic[512 /* MAX_LINE_LEN * 2 */];
+    /* Safety check for NULL message */
+    if (!msg) {
+        printf("[Error]: Unknown error occurred\n");
```

Comment:
What's the purpose of adding a NULL check here?
AFAICS there is no case where a NULL pointer is passed in, so this doesn't address any real security issue.
Also, I don't see a strong reason to change the API to allow NULL pointers as valid input.